From 111d1b508b798fd2fdc2ca1555f774bc7b7a2534 Mon Sep 17 00:00:00 2001
From: Michal Simek <michal.simek@xilinx.com>
Date: Wed, 28 Nov 2012 17:13:17 +0100
Subject: [PATCH] mtd: xilinx_nandps: Clean coding style issue

Split testing cases from not good looking if else structures.

Signed-off-by: Michal Simek <michal.simek@xilinx.com>
Acked-by: John Linn <john.linn@xilinx.com>
---
 drivers/mtd/nand/xilinx_nandps.c |    9 +++++----
 1 files changed, 5 insertions(+), 4 deletions(-)

diff --git a/drivers/mtd/nand/xilinx_nandps.c b/drivers/mtd/nand/xilinx_nandps.c
index d1fa453..5f8250a 100644
--- a/drivers/mtd/nand/xilinx_nandps.c
+++ b/drivers/mtd/nand/xilinx_nandps.c
@@ -382,11 +382,12 @@ int xnandps_correct_data(struct mtd_info *mtd, unsigned char *buf,
 		/* Toggling error bit */
 		buf[byte_addr] ^= (1 << bit_addr);
 		return 1;
-	} else if (onehot(ecc_odd | ecc_even) == 1) {
-		return 1; /* one error in parity */
-	} else {
-		return -1; /* Uncorrectable error */
 	}
+
+	if (onehot(ecc_odd | ecc_even) == 1)
+		return 1; /* one error in parity */
+
+	return -1; /* Uncorrectable error */
 }
 
 /**
-- 
1.7.1

